cs/src/comm/epoxy-transport/EpoxyTransport.cs

```diff
@@ -395,60 +406,25 @@ public static IPEndPoint ParseStringAddress(string address)
-                // the following on Windows:
-                //
-                // struct tcp_keepalive
-                // {
-                //     u_long onoff; // 0 for off, non-zero for on
-                //     u_long keepalivetime; // milliseconds
-                //     u_long keepaliveinterval; // milliseconds
-                // };
-                //
-                // On some platforms this gets mapped to the relevant OS structures, but on other
-                // platforms, this may fail with a PlatformNotSupportedException.
-                UInt32 keepAliveTimeMillis = checked((UInt32) timeoutConfig.KeepAliveTime.TotalMilliseconds);
-                UInt32 keepAliveIntervalMillis = checked((UInt32) timeoutConfig.KeepAliveInterval.TotalMilliseconds);
-
-                var keepAliveVals = new byte[sizeof (UInt32)*3];
-                keepAliveVals[0] = 1;
-
-                keepAliveVals[4] = (byte) (keepAliveTimeMillis & 0xff);
-                keepAliveVals[5] = (byte) ((keepAliveTimeMillis >> 8) & 0xff);
-                keepAliveVals[6] = (byte) ((keepAliveTimeMillis >> 16) & 0xff);
-                keepAliveVals[7] = (byte) ((keepAliveTimeMillis >> 24) & 0xff);
-
-                keepAliveVals[8] = (byte) (keepAliveIntervalMillis & 0xff);
-                keepAliveVals[9] = (byte) ((keepAliveIntervalMillis >> 8) & 0xff);
-                keepAliveVals[10] = (byte) ((keepAliveIntervalMillis >> 16) & 0xff);
-                keepAliveVals[11] = (byte) ((keepAliveIntervalMillis >> 24) & 0xff);
-
-                try
-                {
-                    socket.IOControl(IOControlCode.KeepAliveValues, keepAliveVals, null);
-                }
-                catch (ObjectDisposedException)
-                {
-                    // Oh well: the connection went down before we could configure it. Nothing to be
-                    // done, except to wait for the next socket operation to fail and let normal
-                    // clean up take over.
-                }
-                catch (Exception ex) when (ex is SocketException || ex is PlatformNotSupportedException)
-                {
-                    logger.Site().Warning(ex, "Socket keep-alive could not be configured");
-                }
+                var socket = new Socket(ipAddress.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
+                await Task.Factory.FromAsync(
+                    socket.BeginConnect,
+                    socket.EndConnect,
+                    ipAddress,
+                    endpoint.Port,
+                    state: null);
+
+                logger.Site().Information(
+                    "Established TCP connection to {0} at {1}:{2}",
+                    endpoint.Host, ipAddress, endpoint.Port);
+                return socket;
+            }
+            catch (SocketException ex)
+            {
+                logger.Site().Error(ex, "Failed to establish TCP connection to {0}", endpoint);
```

Comment:
See whether this can be removed. There should be a top-level try/catch that will log this.